src/allmydata/test/web/test_web.py

```diff
@@ -1035,6 +1035,122 @@ def _check_retrieve(res):
 
         return d
 
+    def test_status_path_error(self):
+        # Expect an error, because path is expected to be of the form
+        # "/status/{up,down,..}-%number", with a hyphen.
+        return self.shouldFail2(error.Error,
+                                "test_status_path_error",
+                                "400 Bad Request",
+                                "no '-' in 'nodash'",
+                                self.GET,
+                                "/status/nodash")
+
+    def test_status_path_404(self):
+        # The test suite (hopefully!) would not have done any setup
+        # for a very large number of statuses at this point, so these
+        # all should return 404.
```

Comment:
Hm it seems like no real statuses are ever created at all.  `WebMixin` makes a `FakeClient` which has a `FakeHistory` which has a hard-coded single status of each type.  Have I followed what's going on here correctly?

Reply:
Ah, yes, my language is clumsy there. :-)

I meant to say that future changes in the test suite will not add a large number of statuses, thus breaking the assumption that there won't be a large number of statuses.

The actual point of this test is to exercise all the iterators in `web.status.Status.getChild()`, which is what codecov.io found [lacking in test coverage](https://codecov.io/gh/tahoe-lafs/tahoe-lafs/compare/354e994f6d00f13e8f877550c1d606454bccb145...144001f3808c86bfdc22cfe2b5c28510d3321591/diff#D1-1226) about this particular PR.  Specifically, this part of `getChild()`:

https://github.com/tahoe-lafs/tahoe-lafs/blob/54e952ceda381a0bef6ee27bc9457b83f10a7537/src/allmydata/web/status.py#L1226-L1248

I will update the comment to clarify this intent.

Reply:
Updated comment with clarification about the real purpose of the test in e48d930.